llvm/lib/Support/StringExtras.cpp

```diff
@@ -96,18 +97,13 @@ std::string llvm::convertToSnakeFromCamelCase(StringRef input) {
   if (input.empty())
     return "";
 
-  std::string snakeCase;
-  snakeCase.reserve(input.size());
-  for (char c : input) {
-    if (!std::isupper(c)) {
-      snakeCase.push_back(c);
-      continue;
-    }
-
-    if (!snakeCase.empty() && snakeCase.back() != '_')
-      snakeCase.push_back('_');
-    snakeCase.push_back(llvm::toLower(c));
+  std::string snakeCase = input.str();
+  for (int i = 0; i < 10; ++i) {
```

Comment:
`.sub` will only substitute left most occurrence.

Reply:
How about looping until the `sub` result's length stops changing?

Reply:
I mean I thought about doing a fixed-point type thing but `while` are strongly discouraged right?